### PR TITLE
show errors for /search (#13, #14)

### PR DESCRIPTION
**Pull request by traumschule**
see #13, #14

- works with autogestion/pubgate/pull/45

### CODE REVIEW

**Comment by traumschule** [comment]
backend throws 400 and 500 with a local http test instance trying to follow @x from @y after looking it up on /search

File: philip/src/components/Search.svelte [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <script>
   export let session;
   export let curRoute;
+  const protocol = base_url.match(/^https/) ? "https" : "http";
```
Comment (autogestion) [inline philip/src/components/Search.svelte:4]: cool)

File: philip/src/components/Search.svelte [hunk 2]
```diff
@@ -10,75 +11,88 @@
...
     let pair = username.split("@");
-    let profile_url = "https://" + pair[1] + "/@" + pair[0];
-    let collection;
+    if (pair.length !== 2) {
+      return (error = "Use this format: username@domain");
+    }
+    let profile_url = `${protocol}://${pair[1]}/@${pair[0]}`;
+
     if (pubgate_instance) {
-      const profile = await fetch(base_url + "/proxy", {
```
Comment (autogestion) [inline philip/src/components/Search.svelte:27]: what do you think about checking domain, and if it is same as `base_url`, do not use proxy and make direct request to API

Comment (autogestion) [reply philip/src/components/Search.svelte:27]: maybe factor out it in separate request util
like this one https://github.com/autogestion/pubgate-philip/blob/master/philip/src/utils/xhr.js

Comment (traumschule) [reply philip/src/components/Search.svelte:27]: 1) good idea, will give it a shot
2) this was for #22? fixed in dbe1720

**Comment by autogestion** [comment]
actually, it do not show the error, just changes button to "Unfollow". That's fine for now. I'll elaborate more on this at #13 
Also Unfollow button do not work as expected. It do sends unfollow object which looks fine, but actually followed actor remains followed. Could be a backend issue

**Pull request merged**